edx-olive/edx-platform
# [SE-2888] Backported changes to enable discussion notifications

This PR backports the changes necessary to enable discussion notifications feature. There are a lot of changes surrounding the discussion notifications but they are not compatible with gingko version and lead to a lot of errors. This only adds the absolute bare minimum changes required without which the feature does not work.

**JIRA tickets:** [SE-2888](https://tasks.opencraft.com/browse/SE-2888), CAM2-341

**Testing Instructions:**
1. Login to LMS with a user account and navigate to discussions tab of a particular course that the user is enrolled in.
2. Enable the checkbox is selected in the `Receive updates` section of the discussion page.
3. Now add a new post but clicking on `Add a Post` button in the top of the page.
4. Logout and login with a second user account.
5. Navigate to the dicussions tab of the previous course and click on `All Discussions` to see the post created earlier.
6. Post a reply on the thread.
7. Verify the first user receives the email notification.

**Reviewers:**


**Authors notes and concerns:**

This also requires the settings `enable_forum_notifications` to set to True in django admin SiteConfiguration object. See [instructions](https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/enable_discussion_notifications.html#enable-discussion-notifications) for more details.
I've mainly been using stage to make and test the changes. But it was leading to some issues while loading studio static assets. I did not debug it to see what was causing the issues but there's a new deployment in Asgard (v106) that works fine. I've been switching it out with the old one (v105) where my changes were deployed, whenever I had to test.
